windows/RNCNetInfo/DefaultNetworkInformation.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Windows.Networking.Connectivity;
+
+namespace ReactNativeCommunity.NetInfo
+{
+    class DefaultNetworkInformation : INetworkInformation
+    {
+        public event NetworkStatusChangedEventHandler NetworkStatusChanged;
+
+        public void Start()
+        {
+            NetworkInformation.NetworkStatusChanged += OnNetworkStatusChanged;
+        }
+
+        public void Stop()
+        {
+            NetworkInformation.NetworkStatusChanged -= OnNetworkStatusChanged;
+        }
+
+        public IConnectionProfile GetInternetConnectionProfile()
+        {
+            var profile = NetworkInformation.GetInternetConnectionProfile();
+            return profile != null
+                ? new ConnectionProfileImpl(profile)
+                : null;
+        }
+
+        private void OnNetworkStatusChanged(object sender)
+        {
+            NetworkStatusChanged?.Invoke(sender);
+        }
+
+        class ConnectionProfileImpl : IConnectionProfile
+        {
+            private readonly ConnectionProfile _profile;
+
+            public ConnectionProfileImpl(ConnectionProfile profile)
+            {
+                _profile = profile;
+            }
+
+            public NetworkConnectivityLevel ConnectivityLevel
+            {
+                get
+                {
+                    return _profile.GetNetworkConnectivityLevel();
+                }
+            }
+
+            public NetworkConnectionType ConnectionType
+            {
+                get
+                {
+                    if (_profile.IsWlanConnectionProfile)
+                    {
+                        return NetworkConnectionType.Wifi;
+                    }
+                    else if (_profile.IsWwanConnectionProfile)
+                    {
+                        return NetworkConnectionType.Cellular;
+                    }
+                    else
+                    {
+                        // If we have a connection that is not a Wifi connection or a Cellular
+                        // connection, then let's fallback and assume it's an Ethernet connection
+                        return NetworkConnectionType.Ethernet;
```

Comment:
We are using the `other` type on Android for when we have an active connection but we don't know what type. Maybe `ethernet` does make more sense on Windows, but if we could detect that then it would be better.

Reply:
I've done some more research and come across a more exact way of detecting an Ethernet connection. However, my laptop doesn't have an Ethernet port so I can't fully test it